### PR TITLE
Java-15020:  Changes made for upgrading DNS round robin test case to …

### PR DESCRIPTION
**Pull request by Bipinkumar27**
…httpclient5

### CODE REVIEW

File: apache-httpclient4/pom.xml [hunk 1]
```diff
@@ -189,6 +189,11 @@
             <version>${spring.version}</version>
             <scope>test</scope>
         </dependency>
```
Comment (anastasiosioannidis) [inline apache-httpclient4/pom.xml:191]: please remove this v5 dependency. In this module we only handle v4

File: apache-httpclient4/pom.xml [hunk 2]
```diff
@@ -289,6 +294,7 @@
         <wiremock.version>2.5.1</wiremock.version>
         <httpcore.version>4.4.16</httpcore.version>
         <httpclient.version>4.5.14</httpclient.version>
```
Comment (anastasiosioannidis) [inline apache-httpclient4/pom.xml:296]: Remove this also

File: apache-httpclient4/src/test/java/com/baeldung/httpclient/HttpAsyncClientV5LiveTest.java
Comment (anastasiosioannidis) [inline apache-httpclient4/src/test/java/com/baeldung/httpclient/HttpAsyncClientV5LiveTest.java:1]: Delete this class and move it to apache-httpclient-2.
In this module we only handle v5 code.

File: apache-httpclient4/src/test/java/com/baeldung/httpclient/HttpAsyncClientV5LiveTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            .build();
+        CloseableHttpClient client = HttpClientBuilder.create()
+            .setDefaultRequestConfig(config)
+            .build();
+
+        ClassicHttpRequest request = ClassicRequestBuilder.get("http://www.google.com:81")
+            .build();
+
+        assertThrows(SocketTimeoutException.class, () -> {
+            client.execute(request, response -> response);
```
Comment (anastasiosioannidis) [inline apache-httpclient4/src/test/java/com/baeldung/httpclient/HttpAsyncClientV5LiveTest.java:35]: Replace "client.execute(request, response -> response); "  with:

client.execute(request, response -> { 
     // handle response 
     return response; 
  }
);

Format above

Comment (Bipinkumar27) [reply apache-httpclient4/src/test/java/com/baeldung/httpclient/HttpAsyncClientV5LiveTest.java:35]: correct me if I am wrong it is doing the same with a lambda expression. Although I have made the suggested changes in https://github.com/eugenp/tutorials/pull/14293

File: apache-httpclient4/pom.xml (hunk 2 above)
Comment (Bipinkumar27) [reply apache-httpclient4/pom.xml:296]: closed this one as created new pr - https://github.com/eugenp/tutorials/pull/14293

File: apache-httpclient4/src/test/java/com/baeldung/httpclient/HttpAsyncClientV5LiveTest.java (hunk 3 above)
Comment (anastasiosioannidis) [reply apache-httpclient4/src/test/java/com/baeldung/httpclient/HttpAsyncClientV5LiveTest.java:35]: Yes, it just seems more natural to the reader. Thanks.

File: apache-httpclient4/src/test/java/com/baeldung/httpclient/HttpAsyncClientV5LiveTest.java
Comment (anastasiosioannidis) [reply apache-httpclient4/src/test/java/com/baeldung/httpclient/HttpAsyncClientV5LiveTest.java:1]: @Bipinkumar27 I made a mistake about the apache-httpclient. Should have been httpclient-simple. Sorry for this. I will comment also in your new PR

**Pull request closed without merging**